src/main/java/org/killbill/billing/plugin/adyen/api/AdyenPaymentPluginApi.java

```diff
@@ -481,15 +487,25 @@ private PaymentInfo buildPaymentInfo(final Account account, final UUID kbPayment
         final String ccType = PluginProperties.getValue(PROPERTY_CC_TYPE, paymentMethodsRecordCcType, properties);
         final String recurringDetailId = PluginProperties.findPluginPropertyValue(PROPERTY_RECURRING_DETAIL_ID, mergedProperties);
 
+        final PaymentInfo paymentInfo;
         if (recurringDetailId != null) {
-            return buildRecurring(paymentMethodsRecord, paymentProvider, mergedProperties);
+            paymentInfo = buildRecurring(paymentMethodsRecord, paymentProvider, mergedProperties);
         } else if ("sepadirectdebit".equals(ccType)) {
-            return buildSepaDirectDebit(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
+            paymentInfo = buildSepaDirectDebit(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
         } else if ("elv".equals(ccType)) {
-            return buildElv(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
+            paymentInfo = buildElv(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
         } else {
-            return buildCreditCard(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
+            paymentInfo = buildCreditCard(nonNullPaymentMethodsRecord, paymentProvider, mergedProperties);
         }
+
+        paymentInfo.setContinuousAuthenticationEnabled(isContinuousAuthenticationEnabled(mergedProperties));
+
+        return paymentInfo;
+    }
+
+    private boolean isContinuousAuthenticationEnabled(final Iterable<PluginProperty> mergedProperties) {
+        final String contAuth = PluginProperties.findPluginPropertyValue(PROPERTY_CONTINUOUS_AUTHENTICATION, mergedProperties);
+        return Boolean.parseBoolean(contAuth);
```

Comment:
After thinking about it and re-reading [adyens documentation](https://docs.adyen.com/display/TD/Recurring+Manual) there are 3 different use cases:

1. Use adyens recurring payments feature with contract "recurring": Cvv is not required (it's an implicit "contAuth") 
2. Use adyens recurring payment feature with contract "one-click": Cvv is always required
3. Use you own card on file system + "contAuth" to simulate option 1. Instead of providing adyens recurringDetailId the merchant retrieves stored payment data from it's store and populates the fields like a normal payment request. "contAuth" is needed to turn adyens (not needed) validations off.

So what we can do is validate that "contAuth" is in fact never used together with a "recurringId".

(btw. we have a flaw in the plugin implementation as we always create ["RECURRING,ONECLICK"](https://github.com/killbill/killbill-adyen-plugin/blob/26e4ad4e6c491db5baad7036de814a648d185095/src/main/java/org/killbill/billing/plugin/adyen/client/payment/builder/PaymentRequestBuilder.java#L40) contracts, but that should be an explicit decision).